src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,119 @@
+import { User, UserWithoutId } from './interfaces/user';
+import { UserParameters } from './interfaces/parameters';
+import { userDocToUser } from '../../mappers/user.mapper';
+import { connection } from '../../main';
+
+class UserRepository {
+  async findByEmail(email: User['email']): Promise<User> {
+    return new Promise((resolve, reject) => {
+      connection.query(`SELECT * FROM users WHERE email = '${email}'`, function (error, results, fields) {
```

Comment:
in real project we can't use creating of query in this way, I mean that we have to hide parameters with `?`
like `SELECT * FROM users WHERE email =  ?, [email]`

Reply:
better to fix it